packages/flutter/lib/src/painting/colors.dart

```diff
@@ -460,3 +460,46 @@ class ColorSwatch<T> extends Color {
   @override
   String toString() => '$runtimeType(primary value: ${super.toString()})';
 }
+
+/// [DiagnosticsProperty] that has an [Color] as value.
+class ColorDiagnosticsProperty extends DiagnosticsProperty<Color> {
+  /// Create a diagnostics property for [Color].
+  ///
+  /// The [showName], [style], and [level] arguments must not be null.
+  ColorDiagnosticsProperty(
+    String name,
+    Color value, {
+      String description,
+      String ifNull,
```

Comment:
I suspect a number of these named arguments are never used for a Color property. For example: users shouldn't be including a custom description or ifNull string for a Color property. If someone ever wanted that functionality they could always add it later as a non-breaking change.

For this and the other property you added we need to also update the generic documentation for adding properties to clarify that these properties NEED to be used instead of DiagnosticsProperty<FOO> for the Color and Icon case.
An alternate perhaps simpler solution would be to give the Delegate a hook to add special JSON for these two cases that way users wouldn't have to use these two classes and could keep things simpler.

Reply:
On the other hand, this code is a bit cleaner although it puts a higher tax on users implementing debugFillProperties.

Reply:
Removed all unused arguments.

Yeah, I also find this code a little bit cleaner and for the espresso use case I also want these additional values in the json output. That's why I think adding them through the delegate will just lead to duplication... I'll update the documentation.